storage/innobase/include/dict0crea.inl

```diff
@@ -54,7 +54,7 @@ dict_create_add_foreign_id(
 		if (dict_table_t::is_temporary_name(name)) {
 
 			/* no overflow if number < 1e13 */
-			sprintf(id, "%s_ibfk_%lu", name,
+			snprintf(id, sizeof(namelen + 20), "%s_ibfk_%lu", name,
```

Comment:
namelen is ulint type (=strlen(name). You probably want to write snprintf(id, namelen + 20, "%s_ibfk_%lu", name, ulong) (*id_nr)++); ?